aten/CMakeLists.txt

```diff
@@ -10,6 +10,34 @@ project(ATen)
 
 cmake_policy(SET CMP0012 NEW)
 
+# ---[ If running on Ubuntu, check system version and compiler version.
+if(EXISTS "/etc/os-release")
+  execute_process(COMMAND
+    "sed" "-ne" "s/^ID=\\([a-z]\\+\\)$/\\1/p" "/etc/os-release"
+    OUTPUT_VARIABLE OS_RELEASE_ID
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+  execute_process(COMMAND
+    "sed" "-ne" "s/^VERSION_ID=\"\\([0-9\\.]\\+\\)\"$/\\1/p" "/etc/os-release"
+    OUTPUT_VARIABLE OS_RELEASE_VERSION_ID
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+  if(OS_RELEASE_ID STREQUAL "ubuntu")
+    if(OS_RELEASE_VERSION_ID VERSION_GREATER "17.04")
+      if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "6.0.0"
+          AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.0.0")
+            message(FATAL_ERROR
+              "Do not use GCC 5. GCC 5 has a known bug in Ubuntu 17.04"
```

Comment:
shouldn't this be `17.10` and not `17.04`?

Reply:
You are right. It was a typo. Fixed it.